File: frontend/demo/component/avatar/avatar-abbreviation.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/avatar';
import '@vaadin/horizontal-layout';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('avatar-abbreviation')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-avatar name="Augusta Ada King"></vaadin-avatar>

        <vaadin-avatar name="Augusta Ada King" abbr="AK"></vaadin-avatar>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarAbbreviation.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.avatar.Avatar;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

@Route("avatar-abbreviation")
public class AvatarAbbreviation extends Div {

  public AvatarAbbreviation() {
    // tag::snippet[]
    Avatar avatarName = new Avatar("Augusta Ada King");

    Avatar avatarAbbr = new Avatar("Augusta Ada King");
    avatarAbbr.setAbbreviation("AK");
    // end::snippet[]

    add(avatarName, avatarAbbr);
  }
  public static class Exporter extends DemoExporter<AvatarAbbreviation> {} // hidden-source-line
}
